---
title: "BCC pre-processing"
author: "Stephen Coleman"
date: "10/02/2020"
output: pdf_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
### Adapted from the data processing file accompanying:
### Lock, E. F., & Dunson, D. B. (2013). Bayesian consensus clustering. Bioinformatics (Oxford, England), 29(20), 2610â€“2616. http://doi.org/10.1093/bioinformatics/btt425


# setwd("~/Dropbox/GwenDataIntegration/BCC")
# source("https://bioconductor.org/biocLite.R")

# if (!requireNamespace("BiocManager", quietly = TRUE))
# install.packages("BiocManager")

# BiocManager::install("impute")

library(impute)
library("MCMCpack")
library(magrittr)
library(stringr)
library(umap)
library(ggfortify)
library(ggplot2)

makeUMAPLabels <- function(umap_mat){
  
  x_med <- median(umap_mat[, 1])
  y_med <- median(umap_mat[, 2])
  
  my_labels <- rep(0, nrow(umap_mat))
  names(my_labels) <- row.names(umap_mat)
  
  my_labels[which(umap_mat[,1] > x_med & umap_mat[,2] > y_med)] <- 1
  my_labels[which(umap_mat[,1] > x_med & umap_mat[,2] <= y_med)] <- 2
  my_labels[which(umap_mat[,1] <= x_med & umap_mat[,2] <= y_med)] <- 3
  my_labels[which(umap_mat[,1] <= x_med & umap_mat[,2] > y_med)] <- 4

  my_labels <- as.factor(my_labels)
  
  my_labels
}

makeUMAPPlotData <- function(umap_mat, labels){
  
  plt_data <- data.frame(x = umap_mat[,1],
                         y = umap_mat[,2],
                         labels = labels
  )
  
  plt_data
}

plotUMAP <- function(plt_data){
  
  p <- ggplot(data = plt_data, aes(x = x, y = y, colour = labels)) +
    geom_point()
  
  p
}


set.seed(1)

### Data can be downloaded from the TCGA Data Portal at: https://tcga-data.nci.nih.gov/docs/publications/brca_2012/

# Create file names
save_loc <- "./"
file_type <- ".png"
plots <- c("_pre", "_post")

data_types <- c(
  "gene_expression",
  "methylation",
  "miRNA",
  "protein"
)

file_names <- lapply(plots, function(x){
  paste0(save_loc, data_types, x, file_type)
})

umap_file_names <- paste0(save_loc, data_types, "_umap", file_type)

```

## Data

Read in the data and check the format:

```{r read_in_data}

data_loc <- "../Raw_data/"
filenames <- c(
  "BRCA.exp.348.med.csv",
  "BRCA.348.precursor.txt",
  "rppaData-403Samp-171Ab-Trimmed.txt",
  "BRCA.Methylation.574probes.802.txt"
)

GE <- read.csv(paste0(data_loc, "BRCA.exp.348.med.csv"), header = TRUE)
miRNA <- read.csv(paste0(data_loc, "BRCA.348.precursor.txt"), header = TRUE)
Protein <- read.table(paste0(data_loc, "rppaData-403Samp-171Ab-Trimmed.txt"), header = TRUE)
Meth <- read.table(paste0(data_loc, "BRCA.Methylation.574probes.802.txt"), header = TRUE)

orig_datafiles <- list(GE, miRNA, Protein, Meth) %>%
  magrittr::set_names(c("GE", "miRNA", "Protein", "Meth"))

dim(GE)
dim(miRNA)
dim(Protein)
dim(Meth)

# View the head of the data
head(GE[, 1:5])
head(miRNA[, 1:5])
# head(Protein[, 1:5])
# head(Meth[, 1:5])
```

For integrative clutstering we need common samples across datasets. Find these.

```{r names}
# Match columns (samples) between sources
namesExp <- names(GE)[2:349]
namesmiRNA <- names(miRNA)[2:349]
namesProtein <- names(Protein)[2:404]
namesMeth <- names(Meth)

head(namesExp)
head(namesProtein)

# Matching samples present
namesExp <- substr(namesExp, 1, 16)
namesmiRNA <- substr(namesmiRNA, 1, 16)
namesProtein <- substr(namesProtein, 1, 16)

MatchProt <- match(namesExp, namesProtein, nomatch = 0)
MatchMeth <- match(namesExp, namesMeth, nomatch = 0)
```

Convert to matrix format and set row names

```{r convert_format}
miRNA_names <- miRNA[, 1]
miRNA <- miRNA[, 2:349]
miRNA.mat <- as.matrix(miRNA[, order(namesmiRNA)]) %>%
  set_rownames(miRNA_names)

Protein.mat <- Protein[, 2:404]
Protein.mat <- as.matrix(Protein.mat[, MatchProt]) %>%
  set_rownames(Protein[, 1])

Meth.mat <- as.matrix(Meth[, MatchMeth]) %>%
  set_rownames(row.names(Meth))

Exp.mat      <- as.matrix(GE[,2:349])

reduced_matrix_data <- list(Exp.mat, miRNA.mat, Protein.mat, Meth.mat) %>% 
  magrittr::set_names(c("GE", "miRNA", "Protein", "Meth"))

```

How much missingness is present?

```{r missingness}

# How many missing entries in our data
print(lapply(reduced_matrix_data, function(x){
  sum(is.na(x))
}
))

# The only dataset with NAs is the Gene expression data
dim(reduced_matrix_data$GE)
ge_missingness <- rowSums(is.na(reduced_matrix_data$GE))
ge_missingness[ge_missingness > 0] %>% 
  hist(main = "Count of NAs in each gene in GE data (0's excluded)",
       xlab = "Number of NAs present")
```

As the Gene Expression contains actual NA's we will impute missing values using ``knn`` with $k=10$ (the default setting).

```{r impute_ge, message=FALSE, warning=FALSE, results="hide"}
# Impute missing values via KNN (K=10) for the Gene expression data
Exp.mat      <- impute.knn(Exp.mat)
Exp.mat      <- Exp.mat$data %>% 
  set_rownames(GE[,1])

reduced_matrix_data$GE <- Exp.mat
```

We might also be interested in the number of 0's present in the datasets as this could represent missing data.

```{r miRNA_missing}
# Check how many 0's there are in the datasets (possibly these are missing points too!)
print(lapply(reduced_matrix_data, function(x){
  sum(x == 0)
}
))

# The GE and miRNA datasets both have 0 entries, check how many genes have 0's
print(lapply(reduced_matrix_data, function(x){
  sum(rowSums(x == 0) > 0)
}
))

ge_zeroness <- rowSums(reduced_matrix_data$GE == 0)
ge_zeroness[ge_zeroness > 0] %>% 
  hist(main = "Count of 0's in each gene in GE data (0's excluded)",
       xlab = "Number of 0's present")

miRNA_zeroness <- rowSums(reduced_matrix_data$miRNA == 0)
miRNA_zeroness %>% 
  hist(main = "Count of 0's in each gene in miRNA data",
       xlab = "Number of 0's present")

miRNA_zeroness[miRNA_zeroness > 0] %>% 
  hist(main = "Count of 0's in each gene in miRNA data (0's excluded)",
       xlab = "Number of 0's present")

# There's a far greater number of 0 entries in the miRNA compared to the GE; 
# also, no gene in the GE dataset has more than 3 associated 0 entries
```

Now we visualise the effect of the preprocessing steps by mapping the points to a 2D UMAP; colouring based upon their UMAP coordinates. We split the points based upon the median of the points in each direction, so points are assigned values based upon:

```{r label_table, echo = FALSE, results = "asis"}
x <- 1:4
y <- c(rep("Greater than median", 2), rep("Less than median", 2))
z <- rep(c("Greater than median", "Less than median"), 2)
label_df <- data.frame("Condition in UMAP 1" = y, "Condition in UMAP 2" = z, "Label" = x)
knitr::kable(label_df, caption = "Labels defined by UMAP coordinates.")
```

First consider the 

```{r ge_preprocessing_visualisation}
# Do PCA
ge_pca <- prcomp(Exp.mat)

# Visaulise the first two components
# autoplot(ge_pca)

# Apply UMAP to the data
ge_umap <- umap(Exp.mat)

# Find labels from the UMAP to track how transform changes layout of data
ge_labels <- makeUMAPLabels(ge_umap$layout)
ge_plt_data <- makeUMAPPlotData(ge_umap$layout, ge_labels)
plotUMAP(ge_plt_data)  +
  labs(title = "Gene expression: pre-processing UMAP",
       subtitle = "Coloured by UMAP coordinates",
       x = "UMAP 1",
       y = "UMAP 2") 

# Save UMAP plot
# ggsave(umap_file_names[1])

# Plot PCA with UMAP defined labels
autoplot(ge_pca, data = Exp.mat, colour = ge_labels) +
  labs(title = "Gene expression: pre-processing",
       subtitle = "Coloured by UMAP coordinates")

# ggsave(file_names[[1]][1])

Exp.mat %>% 
  apply(1, sd) %>% 
  hist(main = "Standard deviation of genes in GE data (before processing)")

processedExpression  <- Exp.mat[apply(Exp.mat,1,sd)>1.5,] ###Filter to select only most variable genes
print(ncol(Exp.mat) - ncol(processedExpression))

processedExpression %>% 
  apply(1, sd) %>% 
  hist(main = "Standard deviation of genes in GE data (post processing)")

ge_zeroness %>% 
  hist(main = "Count of 0's in each gene in GE data post-processing",
       xlab = "Number of 0's present")

# Look at PCs after transform with the same labelling
p_ge_pca <- prcomp(processedExpression)
# autoplot(p_ge_pca)
autoplot(p_ge_pca, data = processedExpression, colour = ge_labels[apply(Exp.mat,1,sd)>1.5]) +
  labs(title = "Gene expression: post-processing",
       subtitle = "Coloured by UMAP coordinates")

# ggsave(file_names[[2]][1])

p_ge_umap <- umap(processedExpression)
p_ge_plt_data <- makeUMAPPlotData(p_ge_umap$layout, ge_labels[apply(Exp.mat,1,sd)>1.5])
plotUMAP(p_ge_plt_data)  +
  labs(title = "Gene expression: post-processing UMAP",
       subtitle = "Coloured by UMAP coordinates",
       x = "UMAP 1",
       y = "UMAP 2") 

# ggsave("~/Documents/PhD/Year_1/Consensus_clustering/Analysis/BCC_TCGA_data/Data/gene_expression_umap_post_processing.png")
```

```{r methylation_processing}
meth_pca <- prcomp(Meth.mat)

meth_umap <- umap(Meth.mat)


meth_labels <- makeUMAPLabels(meth_umap$layout)
meth_plt_data <- makeUMAPPlotData(meth_umap$layout, meth_labels)
plotUMAP(meth_plt_data)  +
  labs(title = "Methylation: pre-processing UMAP",
       subtitle = "Coloured by UMAP coordinates",
       x = "UMAP 1",
       y = "UMAP 2") 

# ggsave(umap_file_names[2])


autoplot(meth_pca, data = Meth.mat, colour = meth_labels) +
  labs(title = "Methylation: pre-processing",
       subtitle = "Coloured by UMAP coordinates")

# ggsave(file_names[[1]][2])

Meth.mat %>% 
  apply(1, sd) %>% 
  hist(main = "Methylation: Gene standard deviation (before processing)")

Meth.mat %>% 
  apply(1, mean) %>% 
  hist(main = "Methylation: Gene mean (after processing)")

processedMethylation <- sqrt(Meth.mat)    ##take square root of methylation data
p_meth_pca <- prcomp(processedMethylation)

processedMethylation %>% 
  apply(1, sd) %>% 
  hist(main = "Methylation: Gene standard deviation (after processing)")

processedMethylation %>% 
  apply(1, mean) %>% 
  hist(main = "Methylation: Gene mean (after processing)")

autoplot(p_meth_pca, data = processedMethylation, colour = meth_labels) +
  labs(title = "Methylation: post-processing",
       subtitle = "Coloured by UMAP coordinates")

# ggsave(file_names[[2]][2])


p_meth_umap <- umap(processedMethylation)
p_meth_plt_data <- makeUMAPPlotData(p_meth_umap$layout, meth_labels)
plotUMAP(p_meth_plt_data) +
  labs(y="Petal length (cm)", x = "Sepal length (cm)")
  

```

```{r miRNA_preprocessing}
# Apply PCA for global structure
miRNA_pca <- prcomp(miRNA.mat)

# Apply UMAP to see local structure (should not be strongly affected by the transforms)
miRNA_umap <- umap(miRNA.mat)

# Create labels to record apporximate clustering from UMAP
miRNA_labels <- makeUMAPLabels(miRNA_umap$layout)

# Create a data.frame of UMAP coordinates and labelling
miRNA_plt_data <- makeUMAPPlotData(miRNA_umap$layout, miRNA_labels)

# Plot
plotUMAP(miRNA_plt_data) +
  labs(title = "miRNA: pre-processing UMAP",
       subtitle = "Coloured by UMAP coordinates",
       x = "UMAP 1",
       y = "UMAP 2") 

# ggsave(umap_file_names[3])


# miRNA_exclude <- row.names(miRNA_plt_data)[miRNA_plt_data$x >10]
# autoplot(prcomp(miRNA.mat[! row.names(miRNA.mat) %in% miRNA_exclude, ]))
autoplot(miRNA_pca, data = miRNA.mat, colour = miRNA_labels) + #, scale = 0) +
  labs(title = "miRNA: pre-processing",
       subtitle = "Coloured by UMAP coordinates")  # +
  # xlim(4e4, 5e4) +
  # ylim(5.5e3, 6.1e3)
# miRNA_pca$x[(miRNA_pca$x[,1] < 4e4),1:2]
# 
# summary(miRNA_pca$x[,1:2])
# 
# miRNA_exclude <- row.names(miRNA_pca$x)[(miRNA_pca$x[,1] < 4e4 & miRNA_pca$x[,2] < 6.1e3)]
# indices_to_drop <- ! row.names(miRNA.mat) %in% miRNA_exclude
# miRNA_reduced <- miRNA.mat[which(indices_to_drop),]
# miRNA_labels_reduced <- miRNA_labels[which(indices_to_drop)]
# 
# miRNA_pca_red <- prcomp(miRNA_reduced)
# 
# autoplot(miRNA_pca_red, data = miRNA_reduced, colour = miRNA_labels_reduced) +
#   labs(title = "miRNA: pre-processing",
#        subtitle = "Coloured by UMAP coordinates") 

# Save the PCA plot with UMAP colouring
# ggsave(file_names[[1]][3])


# Transform the data
miRNA_to_drop <- rowSums(miRNA.mat==0) < 348*0.5
miRNA.mat    <- miRNA.mat[which(miRNA_to_drop),]
processedmiRNA       <- log(1+miRNA.mat) ##take log of miRNA data

# Take the PCA of the transformed data
p_miRNA_pca <- prcomp(processedmiRNA)

# Plot with the same labelling as previosuly applied
autoplot(p_miRNA_pca, data = processedmiRNA, colour = miRNA_labels[which(miRNA_to_drop)]) +
  labs(title = "miRNA: post-processing",
       subtitle = "Coloured by UMAP coordinates") 

# ggsave(file_names[[2]][3])

p_miRNA_umap <- umap(processedmiRNA)
p_miRNA_plt_data <- makeUMAPPlotData(p_miRNA_umap$layout, miRNA_labels[which(miRNA_to_drop)])
plotUMAP(p_miRNA_plt_data) +
  labs(title = "miRNA: post-processing UMAP",
       subtitle = "Coloured by UMAP coordinates",
       x = "UMAP 1",
       y = "UMAP 2") 

# ggsave("~/Documents/PhD/Year_1/Consensus_clustering/Analysis/BCC_TCGA_data/Data/miRNA_umap_post_processing.png")


# Remove miRNAs with > 50% 0 entries
print(dim(miRNA.mat))
miRNA.mat    <- miRNA.mat[rowSums(miRNA.mat==0) < 348*0.5,]
print(dim(miRNA.mat))

```

```{r protein_processing}
protein_pca <- prcomp(Protein.mat)
autoplot(protein_pca)
protein_umap <- umap(Protein.mat)
plot(protein_umap$layout)

protein_labels <- makeUMAPLabels(protein_umap$layout)
protein_plt_data <- makeUMAPPlotData(protein_umap$layout, protein_labels)
plotUMAP(protein_plt_data) +
  labs(title = "Protein: pre-processing UMAP",
       subtitle = "Coloured by UMAP coordinates",
       x = "UMAP 1",
       y = "UMAP 2") 

# ggsave(umap_file_names[4])

autoplot(protein_pca, data = Protein.mat, colour = protein_labels) +
  labs(title = "Protein: pre-processing",
       subtitle = "Coloured by UMAP coordinates") 

# ggsave(file_names[[1]][4])


processedProtein     <- scale(Protein.mat,center=TRUE,scale=TRUE) #Column center/scale protein
p_protein_pca <- prcomp(processedProtein)
autoplot(p_protein_pca, data = processedProtein, colour = protein_labels)  +
  labs(title = "Protein: post-processing",
       subtitle = "Coloured by UMAP coordinates") 

# ggsave(file_names[[2]][4])


p_protein_umap <- umap(processedProtein)
p_protein_plt_data <- makeUMAPPlotData(p_protein_umap$layout, protein_labels)
plotUMAP(p_protein_plt_data) +
  labs(title = "Protein: post-processing UMAP",
       subtitle = "Coloured by UMAP coordinates",
       x = "UMAP 1",
       y = "UMAP 2") 
```
